controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -96,8 +99,11 @@ func (r *ClowdEnvironmentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		Env:    &env,
 	}
 
+	err = r.generateAppStatus(provider)
```

Comment:
There is a call to `SetDeploymentStatus` later on in this function that populates another part of the status section.  Can you move this call next to that one?  They should probably share the same parameter set as well.  Either way works, just so they're consistent.  Names could be consistent, too -- one starts with `Generate` and the other with `Set`.